cider/function/FunctionLookup.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "function/FunctionLookup.h"
+
+void FunctionLookup::registerFunctionLookUpContext() {
+  // internal scalar function
+  scalar_function_look_up_ptr_map_.insert(
+      std::make_pair<std::string, SubstraitFunctionLookupPtr>(
+          PRESTO_ENGINE,
+          std::make_shared<cider::function::substrait::SubstraitScalarFunctionLookup>(
+              cider_internal_function_ptr_, substrait_mappings_)));
+  scalar_function_look_up_ptr_map_.insert(
+      std::make_pair<std::string, SubstraitFunctionLookupPtr>(
+          SUBSTRAIT_ENGINE,
+          std::make_shared<cider::function::substrait::SubstraitScalarFunctionLookup>(
+              cider_internal_function_ptr_, presto_mappings_)));
+  // internal aggregate function
+  aggregate_function_look_up_ptr_map_.insert(
+      std::make_pair<std::string, SubstraitFunctionLookupPtr>(
+          PRESTO_ENGINE,
+          std::make_shared<cider::function::substrait::SubstraitAggregateFunctionLookup>(
+              cider_internal_function_ptr_, substrait_mappings_)));
+  aggregate_function_look_up_ptr_map_.insert(
+      std::make_pair<std::string, SubstraitFunctionLookupPtr>(
+          SUBSTRAIT_ENGINE,
+          std::make_shared<cider::function::substrait::SubstraitAggregateFunctionLookup>(
+              cider_internal_function_ptr_, presto_mappings_)));
+  // extension function
+  extension_function_look_up_ptr_map_.insert(
+      std::make_pair<std::string, SubstraitFunctionLookupPtr>(
+          PRESTO_ENGINE,
+          std::make_shared<cider::function::substrait::SubstraitScalarFunctionLookup>(
+              substrait_extension_function_ptr_, substrait_mappings_)));
+  extension_function_look_up_ptr_map_.insert(
+      std::make_pair<std::string, SubstraitFunctionLookupPtr>(
+          SUBSTRAIT_ENGINE,
+          std::make_shared<cider::function::substrait::SubstraitScalarFunctionLookup>(
+              presto_extension_function_ptr_, presto_mappings_)));
+}
```

Comment:
So here you register all the maps as the context? The expected APIs are like following:
`FunctionLookup func_lookup = new FunctionLookup (std::string from_platform); 
FunctionDescription func_desc = func_lookup.lookup(function_signature);
`
In this way, we don't need `scalar_function_look_up_ptr_map_` here.

Reply:
Done.